redis/src/main/java/com/yahoo/ycsb/db/RedisClusterClient.java

```diff
@@ -0,0 +1,268 @@
+ */
+
+/**
+ * Redis client binding for YCSB.
+ *
+ * All YCSB records are mapped to a Redis *hash field*. For scanning operations, all keys are saved
+ * (by an arbitrary hash) in a sorted set.
+ */
+
+package com.yahoo.ycsb.db;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.Future;
+import java.util.function.Function;
+
+import com.fabahaba.jedipus.client.RedisClient;
+import com.fabahaba.jedipus.cluster.CRC16;
+import com.fabahaba.jedipus.cluster.Node;
+import com.fabahaba.jedipus.cluster.RedisClusterExecutor;
+import com.fabahaba.jedipus.cluster.RedisClusterExecutor.ReadMode;
+import com.fabahaba.jedipus.cmds.Cmds;
+import com.fabahaba.jedipus.cmds.RESP;
+import com.fabahaba.jedipus.pool.ClientPool;
+import com.fabahaba.jedipus.primitive.RedisClientFactory;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+import com.yahoo.ycsb.StringByteIterator;
+
+/**
+ * YCSB binding for <a href="http://redis.io/topics/cluster-spec">Redis Cluster</a>.
+ *
+ * See {@code redis/README.md} for details.
+ */
+public class RedisClusterClient extends DB {
+
+  private RedisClusterExecutor rce;
+
+  public static final String HOST_PROP = "redis.cluster.host";
+  public static final String PORT_PROP = "redis.cluster.port";
+  public static final String PASSWORD_PROP = "redis.cluster.password";
+
+  public static final String MAX_POOL_SIZE_PROP = "redis.cluster.pool.maxclients";
+
+  public static final String INDEX_KEY = "_indices";
+  private static final byte[] INDEX_KEY_BYTES = RESP.toBytes("_indices");
+  private static final int INDEX_SLOT = CRC16.getSlot(INDEX_KEY_BYTES);
+
+  @Override
+  public void init() throws DBException {
+    final Properties props = getProperties();
+
+    int port;
+    final String portString = props.getProperty(PORT_PROP);
+    if (portString != null) {
+      port = Integer.parseInt(portString);
+    } else {
+      port = 7000;
+    }
+
+    String host = props.getProperty(HOST_PROP);
+    if (host == null) {
+      host = "localhost";
+    }
+
+    final ClientPool.Builder poolBuilder = ClientPool.startBuilding().withBlockWhenExhausted(true);
+
+    final String maxPoolSize = props.getProperty(MAX_POOL_SIZE_PROP);
+    if (maxPoolSize != null) {
+      poolBuilder.withMaxTotal(Integer.parseInt(maxPoolSize));
+    }
+
+    final RedisClientFactory.Builder clientFactory = RedisClientFactory.startBuilding();
+    final String password = props.getProperty(PASSWORD_PROP);
+    if (password != null) {
+      clientFactory.withAuth(password);
+    }
+
+    final Function<Node, ClientPool<RedisClient>> masterPoolFactory =
+        node -> poolBuilder.create(clientFactory.createPooled(node));
```

Comment:
are we jdk8 only due to the jedis dependency or due to the use of jdk8 features here in the binding?